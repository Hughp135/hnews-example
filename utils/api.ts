export const getTopStoryIDs = async (): Promise<number[]> => {
  const res = await fetch(
    "https://hacker-news.firebaseio.com/v0/topstories.json"
  );

  if (!res.ok) {
    throw new Error("Failed to fetch top stories");
  }

  return res.json();
};

export interface Story {
  by: string;
  score: number;
  time: number;
  title: string;
  type: string;
  url: string;
}

export const getStory = async (id: number): Promise<Story> => {
  const res = await fetch(
    `https://hacker-news.firebaseio.com/v0/item/${id}.json`
  );

  if (!res.ok) {
    throw new Error("Failed to fetch story data");
  }

  return res.json();
};
