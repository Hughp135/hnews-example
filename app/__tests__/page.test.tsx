import { render } from "@testing-library/react";
import Home from "../page";

import { getStory, getTopStoryIDs, Story } from "@/utils/api";

jest.mock("@/utils/api");

describe("Page", () => {
  const exampleStory: Story = {
    by: "test",
    score: 1,
    time: 201312,
    title: "A story is here",
    type: "story",
    url: "http://test.url",
  };

  beforeEach(() => {
    (getStory as jest.Mock).mockResolvedValue(exampleStory);
    (getTopStoryIDs as jest.Mock).mockResolvedValue([]);
  });

  it("renders the homepage", async () => {
    const homepageSync = await Home();
    const { asFragment } = render(homepageSync);

    expect(asFragment()).toMatchSnapshot();
  });
});
