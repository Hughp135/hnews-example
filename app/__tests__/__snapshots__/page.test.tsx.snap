// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders the homepage 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <div
      class="infinite-scroll-component__outerdiv"
    >
      <div
        class="infinite-scroll-component "
        style="height: auto; overflow: auto;"
      >
        <h4>
          No more stories to load
        </h4>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
